### PR TITLE
User mutation edge cases

### PR DESCRIPTION
**Pull request by sandisz-d734m37**
## What was changed:
Feat:
Initially, BE planned to build custom error handling to make it easy for FE to handle edge cases pertaining to calls to our service. After much consideration though, BE found GraphQL's built-in error handling to be very similar to and more concise than the custom error handling we had and were trying to build ourselves.

That said, BE had found a few errors GraphQL doesn't know what to do with. In this PR, I've made a fixed for one such case: when a user attempts to update their email address to one already associated with another account, our service will send back an error that reads "<EMAIL INPUT BY USER> is already associated with another account. Your email must be unique."

Refactor:
- `./app/graphql/mutations/edit_user.rb`
- `./spec/mutations/editUser_request_spec.rb`
- Note: all mutation-related spec tests were moved from `./spec/requests` to a new sub-directory, `./spec/requests/mutations`

Test:
- `./spec/mutations/editUser_request_spec.rb`

## Explain the reason for changes: (If there is no spec file how were things tested)
BE wanted to make it easy for FE to handle errors, so I built some custom error handling specifically for user mutations (used for User creation, updates, and deletion). After much consideration, several of these custom errors were removed to let GraphQLs built-in error handling take precedence.

In this PR, you'll find fully-tested, custom error handling specifically for the editUser mutation in the case a user attempts to change their email to one already associated with another account. This was left in because graphQL seemingly didn't know how to handle this edge case.

### CODE REVIEW

File: app/graphql/mutations/edit_user.rb [hunk 1]
```diff
@@ -3,14 +3,17 @@ class EditUser < BaseMutation
...
+    field :user, Types::UserType
+    
     def resolve(id:, **args)
       updated_user = User.find(id)
+
       updated_user.update(args)
       if updated_user.save
         { user: updated_user }
+      else
+        raise GraphQL::ExecutionError.new("#{args[:email]} is already associated with another account. Your email must be unique.")
```
Comment (sandisz-d734m37) [inline app/graphql/mutations/edit_user.rb:15]: This is the real/main change to this file to look out for.

File: spec/requests/mutations/editUser_request_spec.rb [hunk 2]
```diff
@@ -37,4 +36,35 @@
...
+
+    post '/graphql', params: { query: error_mutation }
+
+    json = JSON.parse(response.body, symbolize_names: true)
+
+    error = json[:errors][0]
+
+
+    expect(error[:message]).to eq("#{user_1.email} is already associated with another account. Your email must be unique.")
+  end
```
Comment (sandisz-d734m37) [inline spec/requests/mutations/editUser_request_spec.rb:69]: This is the test for lines 14 + 15 in app/graphql/mutations/edit_user.rb (see comment in file above)

File: app/graphql/mutations/edit_user.rb (hunk 1 above)
Comment (sandisz-d734m37) [reply app/graphql/mutations/edit_user.rb:15]: Meaning the main thing this PR is centered around.

**Review by ShermanA-13 — APPROVED** [review]
looks solid, seems like everything we talked about on Friday was addressed is here.

**Review by phillipkamps — APPROVED** [review]
Sorry that some of your hard work didn't get used. 😞 We appreciate you doing this! 💙

**Pull request merged**